### PR TITLE
add debug section to simulator doc

### PR DESCRIPTION
**Pull request by nlu90**
Add debug docs on how to set breakpoint for simulator and how to save output to local file based on the discussion in #844

### CODE REVIEW

File: website/content/docs/developers/simulator-mode.md [hunk 1]
```diff
@@ -39,8 +39,17 @@ simulator.deactivate("test");
...
+
+The simulator mode will run in separate threads other than main thread. All those interfaces are thread-safe. This means that you could invoke these interfaces in other threads and monitor the corresponding behaviors interactively.
+
+# Debug topology using IntelliJ
+
+Bolts and Spouts run as separate threads in simulator. To add breakpoints inside a bolt/spout, the Suspend Policy of the breakpoint needs to be set as Thread. To change the Suspend Policy, Right click on the breakpoint as shown in the following image:
+
+![Set Breakpoint](/img/intellij-set-breakpoint.jpg)
+
+If output and logs are not convenient to be checked in intellij console, they can be saved to local file by choose `Run -> Edit Configurations....` and set as shown in the following image:
```
Comment (objmagic) [inline website/content/docs/developers/simulator-mode.md]: maybe refactor to 

"If it's not convenient to check the output and logs in intellij console, you can save them to a local file by choosing `Run -> Edit Configurations....` as shown in the following image:"

Comment (nlu90) [inline website/content/docs/developers/simulator-mode.md]: I remember there's a discussion about not using "you" in the doc. I'll change other 'you's occurred in this doc to 'one'

Comment (lucperkins) [inline website/content/docs/developers/simulator-mode.md]: The result of that discussion was that "you" is okay in tutorial-style docs but not quite as welcome in things like REST API docs or Javadoc. In this case, I think it's appropriate.

Comment (nlu90) [inline website/content/docs/developers/simulator-mode.md]: Thanks for the clarification @lucperkins.

Do you think I need to revert the 'you' changes? Or we can just merge this PR here?

File: website/content/docs/developers/simulator-mode.md [hunk 2]
```diff
@@ -39,8 +39,17 @@ simulator.deactivate("test");
...
 ```
 
-To kill a topology, you could also simply terminate this process.
+To kill a topology, one could also terminate the process.
+
+The simulator mode will run in separate threads other than main thread. All those interfaces are thread-safe. This means that one could invoke these interfaces in other threads and monitor the corresponding behaviors interactively.
+
+# Debug topology using IntelliJ
+
+Bolts and Spouts run as separate threads in simulator. To add breakpoints inside a bolt/spout, the Suspend Policy of the breakpoint needs to be set as Thread. To change the Suspend Policy, Right click on the breakpoint as shown in the following image:
```
Comment (billonahill) [inline website/content/docs/developers/simulator-mode.md]: Lowercase 'right click'.

Also here and elsewhere best to line break these at 80 or 100 to make it easier to provide comments and read in an IDE.

File: website/content/docs/developers/simulator-mode.md (hunk 1 above)
Comment (billonahill) [inline website/content/docs/developers/simulator-mode.md]: As refactored, I think this reads well when just desribing how to tools work without saying what "you need to do". For the @objmagic suggestion to be consistent you could remove "you can" from the sentence and it reads well.

Also capitalize intellij as IntelliJ.

File: website/content/docs/developers/simulator-mode.md [hunk 3]
```diff
@@ -2,15 +2,15 @@
...
-
 Simulator mode is specifically designed for topology developers to better debug or optimize their topologies.
 
 Simulator mode simulates a heron cluster in a single JVM process. This is useful for developing and testing topologies.
 Running topologies under simulator mode is similar to running topologies on a cluster.
 
-With a whole topology running in a single process, you could enjoy all free benefits it brings.
-For example, one can run program in IDE and set breakpoints to examine the states of a topology, or profile your program to optimize it.
+With a whole topology running in a single process, one could enjoy all free benefits it brings.
+For example, one can run program in IDE and set breakpoints to examine the states of a topology, or profile the program to optimize it.
```
Comment (billonahill) [inline website/content/docs/developers/simulator-mode.md]: "an IDE"

File: website/content/docs/developers/simulator-mode.md [hunk 4]
```diff
@@ -2,15 +2,15 @@
 title: Simulator Mode
 ---
 
-This page explains how to develop a topology on your local machine.
-
 Simulator mode is specifically designed for topology developers to better debug or optimize their topologies.
```
Comment (billonahill) [inline website/content/docs/developers/simulator-mode.md]: s/better/easily

File: website/content/docs/developers/simulator-mode.md [hunk 5]
```diff
@@ -2,15 +2,15 @@
...
 
 Simulator mode simulates a heron cluster in a single JVM process. This is useful for developing and testing topologies.
```
Comment (billonahill) [inline website/content/docs/developers/simulator-mode.md]: "... in a single JVM process, which is useful ..."

Comment (nlu90) [inline website/content/docs/developers/simulator-mode.md]: Thanks for reviewing this doc. I updated all the comments.

File: website/content/docs/developers/simulator-mode.md [hunk 6]
```diff
@@ -39,8 +40,22 @@ simulator.deactivate("test");
 simulator.shutdown();
 ```
 
-To kill a topology, you could also simply terminate this process.
```
Comment (billonahill) [inline website/content/docs/developers/simulator-mode.md]: "Other interfaces for simulator mode are:" should be " Other lifecycle methods to use with simulator mode are:"

File: website/content/docs/developers/simulator-mode.md [hunk 7]
```diff
@@ -39,8 +40,22 @@ simulator.deactivate("test");
...
+To kill a topology, one could also terminate the process.
+
+The simulator mode will run in separate threads other than main thread. All those interfaces are
```
Comment (billonahill) [inline website/content/docs/developers/simulator-mode.md]: ...the main thread.

What do you mean by "All those interfaces"? It's not clear what you're referring to here? Do you mean to say "The `Simulator` class is thread safe."?

File: website/content/docs/developers/simulator-mode.md [hunk 8]
```diff
@@ -39,8 +40,22 @@ simulator.deactivate("test");
 simulator.shutdown();
 ```
 
-To kill a topology, you could also simply terminate this process.
+To kill a topology, one could also terminate the process.
+
+The simulator mode will run in separate threads other than main thread. All those interfaces are 
+thread-safe. This means that one could invoke these interfaces in other threads and monitor the
```
Comment (billonahill) [inline website/content/docs/developers/simulator-mode.md]: again, what are "these interfaces"? I see methods on a Simulator class discussed above, not interfaces.

File: website/content/docs/developers/simulator-mode.md [hunk 9]
```diff
@@ -39,8 +40,22 @@ simulator.deactivate("test");
 simulator.shutdown();
 ```
 
-To kill a topology, you could also simply terminate this process.
+To kill a topology, one could also terminate the process.
+
+The simulator mode will run in separate threads other than main thread. All the above methods are
```
Comment (billonahill) [inline website/content/docs/developers/simulator-mode.md]: "...than the main thread."

**Comment by billonahill** [comment]
Just left one last small comment. Fix it and ship it.

👍

**Pull request merged**